### DISCUSSION TITLE
Misaligned items

### DISCUSSION THREAD

**Discussion by artelse** [kickingvegas/casual-dired#27] [Q&A] [answered]
Testing out this interesting transient, the menu items don't line up properly. I don't have this issue with Magit for example.
See screenshot:
![Screenshot-2024-05-06-20:44:31](https://github.com/kickingvegas/casual-dired/assets/787801/dbd4f528-2004-47a6-9d07-8f2ee493fe79)

**Comment by kickingvegas** [comment]
Thanks for submitting this issue. A couple of observations:

-   Is your default typeface monospace or fixed-width?
    I currently use a fixed-width typeface and do not see such skewed alignment.
-   I have recently pushed an update [v1.0.3](https://github.com/kickingvegas/casual-dired/releases/tag/1.0.3) which makes default only the usage of ASCII characters for labels.
    UTF-8 symbol support must now be explicitly be enabled from the Settings menu. It should be available on MELPA by end of day (PST) 2024-05-06. With this update, do you have alignment issues? If you are using a variable-width typeface, I would expect so.
-   I would be surprised if Magit has solved horizontal alignment issues with variable-width typefaces.

Thanks again for trying Casual Dired out. I look forward to hearing back from you.

**Comment by artelse** [comment]
Typeface is fixed-pitch, but also have mixed-pitch enabled.
For reference, here's Magit's:
![Screenshot-2024-05-06-22:54:50](https://github.com/kickingvegas/casual-dired/assets/787801/e60950d0-21a5-47db-9d90-ead80b4a12dd)

**Comment by kickingvegas** [comment]
I suspect then it is the UTF-8 symbols whose spacing will be typeface dependent.

Try to update to the v1.0.3 release (it should be available on MELPA about 3 hours from the time of this comment) that now avoids using UTF-8 symbols by default.

**Comment by kickingvegas** [comment]
v1.0.3 now on MELPA. https://melpa.org/#/casual-dired

**Comment by artelse** [comment]
I also suspected the symbols, but it remains misaligned. Is there a column width that throws things off somehow?
![Screenshot-2024-05-07-07:45:51](https://github.com/kickingvegas/casual-dired/assets/787801/6cc847f0-e9a6-4ead-922a-9eecc14a6229)

**Comment by kickingvegas** [comment]
@artelse TBH I'm struggling with this as I am not seeing alignment issues as you can see from the screenshot in the README.org file. https://github.com/kickingvegas/casual-dired/blob/main/README.org

Going to convert this to a discussion item to open this to a wider audience.

**Comment by kickingvegas** [comment]
Are other folks seeing alignment problems with Casual Dired?

**Comment by kickingvegas** [comment]
@artelse perhaps the ellipses which I did not convert from a UTF-8 symbol is throwing it off?

**Reply by artelse** [reply]
Could be, I'll try another font later on, am too busy right now.

**Reply by artelse** [reply, marked as answer]
That was it, I changed font to DejaVu Mono and now it aligns as expected. But, when looking at the screenshot, the ellipses are indeed the real culprit. If that can be fixed, we can use any monospace font.

**Comment by kickingvegas** [comment]
@artelse thanks for confirming the original typeface was not handling ellipses and likely '›' as well. Trying to follow UI conventions that use both symbols. TBH I'm reluctant to design for any monospace typeface as they can vary greatly in quality. My preference is to target typefaces that cover most common glyphs. This includes math symbols, pictographs, lines and boxes.

That said, I appreciate your feedback. I'll consider this matter closed.

**Reply by artelse** [reply]
I was using Prot's build of the `Iosevka Comfy` font. Switching to the `Iosevka Comfy Fixed` variant, which does away with ligatures, now aligns the menus properly.

**Reply by kickingvegas** [reply]
Good to know. IMHO with Unicode and UTF-8 support being far more common, I personally avoid ligatures as they are impossible to distinguish from a Unicode symbol.

**Reply by artelse** [reply]
I've had instances with ligatures in writing of students that caused problems. Wasn't aware the variant I used had ligature support, so now fixed.

**Comment by AndreaOrru** [comment]
I'm also experiencing this with Iosevka Nerd Font, but not with Iosevka Nerd Font Mono.

**Reply by kickingvegas** [reply]
Reopening discussion to show history. Still resolved that this imho is a wontfix issue as this behavior is typeface dependent.

**Comment by woolsweater** [comment]
Anyone having difficulty, try setting `transient-align-variable-pitch` to `t` if you haven't already. It causes columns to be aligned based on the actual pixel size of the content. It is intended for use when your default face is proportional (variable-pitch), but even with a monospace font the emoji/picture glyphs may not be the same width as everything else.
